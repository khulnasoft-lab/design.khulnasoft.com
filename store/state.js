export const getDefaultFrontmatter = () => ({
  attributes: {},
  body: '',
  lastUpdatedAt: '',
});

export default () => ({
  frontmatter: getDefaultFrontmatter(),
});
