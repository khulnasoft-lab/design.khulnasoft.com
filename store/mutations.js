import { getDefaultFrontmatter } from './state';

export default {
  resetFrontmatter(state) {
    state.frontmatter = getDefaultFrontmatter();
  },
  setFrontmatter(state, frontmatter) {
    state.frontmatter = frontmatter;
  },
};
