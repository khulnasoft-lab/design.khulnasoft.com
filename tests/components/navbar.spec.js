import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Navbar from '../../components/navbar.vue';
import mutations from '../../store/mutations';
import state from '../../store/state';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('navbar component', () => {
  let wrapper;
  let store;

  const createComponent = () => {
    store = new Vuex.Store({
      state: state(),
      mutations,
    });
    wrapper = mount(Navbar, {
      localVue,
      store,
      stubs: {
        search: true,
      },
      mocks: {
        $route: {
          fullPath: '/',
          params: {
            section: '',
          },
        },
      },
    });
  };

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders sidebar properly', () => {
    expect(wrapper.find('nav').element).toMatchSnapshot();
  });
});
