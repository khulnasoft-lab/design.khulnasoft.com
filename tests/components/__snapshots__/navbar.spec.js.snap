// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar component renders sidebar properly 1`] = `
<nav
  aria-labelledby="nav-heading"
  class="sidebar__nav"
>
  <h2
    class="gl-sr-only"
    id="nav-heading"
  >
    Main navigation
  </h2>
   
  <ul>
    <li>
      <span
        aria-hidden="true"
        class="sidebar__nav-label gl-font-weight-bold"
        id="item-top1"
      >
        
    Top-level item 1
  
      </span>
       
      <ul
        aria-labelledby="item-top1"
      >
        <li
          class="sidebar__nav-option"
        >
          <nuxt-link-stub
            class="sidebar__nav-anchor"
            to="/top1/second1"
          >
            
    Second-level item 1
  
          </nuxt-link-stub>
        </li>
        <li>
          <button
            aria-controls="item-second2"
            class="sidebar__nav-toggle"
          >
            <span
              class="sidebar__nav-toggle-inner"
            >
              <span>
                Second-level item 2
              </span>
               
              <svg>
                <use
                  xlink:href="#chevron-down"
                />
              </svg>
            </span>
          </button>
           
          <ul
            aria-label="Second-level item 2"
            class="gl-ml-4!"
            id="item-second2"
            style="display: none;"
          >
            <li
              class="sidebar__nav-option"
            >
              <nuxt-link-stub
                class="sidebar__nav-anchor"
                to="/top1/second2/third1"
              >
                
    Third-level item 1
  
              </nuxt-link-stub>
            </li>
            <li
              class="sidebar__nav-option"
            >
              <nuxt-link-stub
                class="sidebar__nav-anchor"
                to="/top1/second2/third2"
              >
                
    Third-level item 2
  
              </nuxt-link-stub>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <span
        aria-hidden="true"
        class="sidebar__nav-label gl-font-weight-bold"
        id="item-top2"
      >
        
    Top-level item 2
  
      </span>
       
      <ul
        aria-labelledby="item-top2"
      >
        <li>
          <button
            aria-controls="item-undefined"
            class="sidebar__nav-toggle"
          >
            <span
              class="sidebar__nav-toggle-inner"
            >
              <span>
                Second-level item without path
              </span>
               
              <svg>
                <use
                  xlink:href="#chevron-down"
                />
              </svg>
            </span>
          </button>
           
          <ul
            aria-label="Second-level item without path"
            class="gl-ml-4!"
            id="item-undefined"
            style="display: none;"
          >
            <li
              class="sidebar__nav-option"
            >
              <nuxt-link-stub
                class="sidebar__nav-anchor"
                to="/top2/third2"
              >
                
    Third-level item 3
  
              </nuxt-link-stub>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;
